pkg/agent/clean/node.go

```diff
@@ -0,0 +1,349 @@
+		logrus.Infof("checking for link %s", l)
+		existing, err := netlink.LinkByName(l)
+		if err != nil {
+			if err.Error() == "Link not found" {
+				logrus.Infof("link %s not found", l)
+				continue // not found, nothing to do
+			}
+
+			return err
+		}
+
+		logrus.Infof("found link and will remove: %s", l)
+		if err := netlink.LinkDel(existing); err != nil {
+			return fmt.Errorf("failed to delete interface: %v", err)
+		}
+		logrus.Infof("link deleted: %s", l)
+	}
+
+	return nil
+}
+
+func Paths() error {
+	logrus.Info("Cleaning up paths...")
+
+	if err := umountTmpfs(); err != nil {
+		return err
+	}
+
+	paths := getPaths()
+	for _, p := range paths {
+		hostPath := path.Join("/host", p)
+		logrus.Infof("trying to delete path: %s", hostPath)
+		_, err := os.Stat(hostPath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				logrus.Infof("path does not exist: %s", hostPath)
+				continue
+			}
+			return err
+		}
+
+		err = os.RemoveAll(hostPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func umountTmpfs() error {
+	// umount tmpfs kubelet stuffs
+	tmpfs, err := exec.Command(
+		"sh", "-c", "mount | grep tmpfs | grep '/var/lib/kubelet' | awk '{ print $3 }'").Output()
+	if err != nil {
+		return err
+	}
+
+	for _, t := range strings.Split(strings.TrimSpace(string(tmpfs)), "\n") {
+		t = strings.TrimSpace(t)
+		if t == "" {
+			continue
+		}
+
+		logrus.Infof("trying to umount tmpfs: %s", t)
+		_, err := exec.Command("sh", "-c", fmt.Sprintf("umount %s", t)).Output()
+		if err != nil {
+			return fmt.Errorf("error trying to umount tmpfs %s: %s", t, err)
+		}
+		logrus.Infof("umount of tmpfs %s successful", t)
+	}
+
+	return nil
+}
+
+func getAgentImage() string {
+	agentImage := os.Getenv("AGENT_IMAGE")
+	if agentImage == "" {
+		agentImage = "rancher/rancher-agent:master"
+	}
+	return agentImage
+}
+
+func cleanDocker(ctx context.Context, c *client.Client) error {
+	blankArgs := filters.NewArgs()
+	if _, err := c.ContainersPrune(ctx, blankArgs); err != nil {
+		return err
+	}
+	if _, err := c.VolumesPrune(ctx, blankArgs); err != nil {
+		return err
+	}
+	if _, err := c.ImagesPrune(ctx, blankArgs); err != nil {
+		return err
+	}
+	if _, err := c.NetworksPrune(ctx, blankArgs); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getPrefixPath() string {
+	return os.Getenv("PREFIX_PATH")
+}
+
+func stopContainers(ctx context.Context, c *client.Client) error {
+	containers, err := c.ContainerList(ctx, types.ContainerListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, container := range containers {
+		isCleanup := false
+		for _, n := range container.Names {
+			if strings.HasPrefix(n, "/"+ContainerName) {
+				isCleanup = true
+				break
+			}
+		}
+
+		if isCleanup {
+			continue // don't stop the cleanup container!
+		}
+
+		config, err := c.ContainerInspect(ctx, container.ID)
+		if err != nil {
+			return err
+		}
+		if strings.HasPrefix(config.Config.Image, "rancher/") {
```

Comment:
Just wanted to clarify; for private registry images, does the image name still start with `rancher/` always? Or could they change the image name as part of mirroring?

Wondering if this should be exposed as an env var option.

Reply:
great question.... all our mirror scripts put them into rancher/* repos as far as i understand... wouldnt mind a second opinion on that tho

Reply:
let's add an env var to let people toggle this, will default to rancher/ but will let people set via CATTLE_REPO_PREFIX